### ISSUE TITLE
Prune latest docs properly 

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#12954]
ref: https://discourse.bokeh.org/t/failed-compilation-of-bokeh-surface3d-example/10184

TLDR; The way we are uploading to `s3://docs.bokeh.org/en/latest/` continually accumulates old cruft from previous releases. These old orphaned files take up space, but also can be confusing for users. 

I've done a manual intervention for now, using the following commands:
```
aws s3 rm s3://docs.bokeh.org/en/latest/ --recursive --only-show-errors

aws s3 cp s3://docs.bokeh.org/en/3.1.0 s3://docs.bokeh.org/en/latest/ --acl bucket-owner-full-control --cache-control max-age=604800,public --recursive --only-show-errors

aws cloudfront create-invalidation --distribution-id E2OC6Q27H5UQ63  --paths "/en/latest*"
```
The release automation can be updated to follow this pattern. I have added `--only-show-errors` which I think may make a non-negligible difference in deploy time, since reporting the "OK" status over the network for every file actually seems rather slow. 

Note: I also reduced cache `max-age` to one week. Rationale is that when `en/latest` is updated, people can see old page versions by default for up to a year unless they manually force-reload. We could make it even shorter, presumably at the expense of a higher AWS expense to some degree. But we should be covered by the NF grant at this time, so maybe thats OK. Should we make the value even smaller for `/en/latest` paths? cc @bokeh/dev
